dashboard/src/components/TableCustomFields.js

```diff
@@ -277,6 +287,34 @@ const EditCustomField = ({ data, editingField, onClose, onSaveField, isNewField,
     onSaveField(editedField);
   };
 
+  const typeIsDisabled = useMemo(() => {
+    if (isNewField || !fieldIsUsed) return false;
+    // the idea to disable the type is to avoid bugs with existing data if the type is changed.
+    // for example, if the type is changed from multi-choice (array) to text (string), or from boolean to text
+    // BUT some data is compatible, let's allow it
+    if (['boolean', 'multi-choice'].includes(field.type)) return true;
+    return false;
+  }, [isNewField, fieldIsUsed, field.type]);
+
+  const optionIsDisabled = useCallback(
+    (option) => {
+      if (typeIsDisabled) return true;
+      if (isNewField) return false;
```

Comment:
```suggestion
      if (isNewField || !fieldIsUsed) return false;
```

J'ai pas testé parce que je suis en train de tenter de lancer mon mobile mais j'ai l'impression qu'on aurait un problème ici : un champ déjà créé mais qui n'est pas utilisé aurait quand même des options grisées ce qui n'est pas normal.